### DISCUSSION TITLE
how do kratos and hydra fit togther?

### DISCUSSION THREAD

**Discussion by nektro** [ory/hydra#2344] [Q&A] [answered]
are you meant to run both, or are they different apps depending on your needs?

**Comment by vinckr** [comment, marked as answer]
They serve very different purposes! 

Both are related to Authentication/Authorization, but 
Ory Hydra is an OAuth 2.0 and OpenID Connect provider.
and 
Ory Kratos is an identity management server.

So think of Ory Kratos as the system that provides login/registration/profile for your users, and all the usual features that come with it. 

Ory Hydra is a bit more complex, but essentially it allows you to become an OAuth2 provider (like for example Google or Github). So you could potentially do your own "Login with Nektro".  
You mostly need it when you have Apps that need machine-2-machine interaction, or enable third-party developers to use your APIs (and pay for it), also for corporate systems ect. The use cases for Ory Hydra are as many as there are for OAuth2. 

Check the [ecosystem documentation](https://www.ory.sh/docs/ecosystem/projects) for more general information on the Ory projects!

**Reply by nektro** [reply]
okay cool, thanks for the response! that's somewhat along the lines of what I was thinking after my first read of the documentation. my only other Q would be is that can I have hydra use a kratos instance as its user database or does it keep one to itself?

**Reply by nektro** [reply]
> ORY Hydra is not identity management, though. Instead, it connects to your existing identity management (e.g. the one from the paragraph above, or ...

sounds great!!
